hw4/hw4.py

```diff
@@ -0,0 +1,55 @@
+import numpy as np
+
+if __name__ == "__main__":
+    arr0 = np.zeros(10)
+    arr1000_7 = np.array([i for i in range(1000, 0, -7)])
+    arr_rand = np.random.rand(10)
+
+
+def multiplication_check(list_of_matrix):
+    a = np.shape(list_of_matrix[0])[0]
+    b = np.shape(list_of_matrix[0])[1]
+    for i in range(len(list_of_matrix)-1):
+        if b != np.shape(list_of_matrix[i+1])[0]:
+            return False
+        else:
+            b = np.shape(list_of_matrix[i+1])[1]
+            a = np.shape(list_of_matrix[i])[0]
+    return True

```

Comment:
Много лишних выражений, можно сильно упростить
```suggestion
    for i in range(len(list_of_matrix)-1):
        if list_of_matrix[i].shape[1] != list_of_matrix[i+1].shape[0]:
            return False
    return True

```